.github/workflows/maven-publish.yml

```diff
@@ -0,0 +1,33 @@
+name: Publish snapshots to maven
+
+on:
+  workflow_dispatch:
+  push:
+      branches:
+        - 'main'
+
+jobs:
+  build-and-publish-snapshots:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write
+      contents: write
+
+    steps:
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin # Temurin is a distribution of adoptium
+          java-version: 11
+      - uses: actions/checkout@v3
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.PUBLISH_SNAPSHOTS_ROLE }}
+          aws-region: us-east-1
+      - name: publish snapshots to maven
+        run: |
+          export SONATYPE_USERNAME=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-username --query SecretString --output text)
+          export SONATYPE_PASSWORD=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-password --query SecretString --output text)
+          echo "::add-mask::$SONATYPE_USERNAME"
+          echo "::add-mask::$SONATYPE_PASSWORD"
+          ./gradlew publish
```

Comment:
Should only publish to snapshots repository

Reply:
@gaiksaya Wondering how is this different from `publishToMavenLocal` ? 

Following https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#testing, set url to 
`/Users/khushbr/test/`
```

% ls /Users/khushbr/test/org/opensearch/performance-analyzer-commons/1.0-SNAPSHOT
maven-metadata.xml						performance-analyzer-commons-1.0-20230523.200158-1.jar		performance-analyzer-commons-1.0-20230523.200158-1.pom
maven-metadata.xml.md5						performance-analyzer-commons-1.0-20230523.200158-1.jar.md5	performance-analyzer-commons-1.0-20230523.200158-1.pom.md5
maven-metadata.xml.sha1						performance-analyzer-commons-1.0-20230523.200158-1.jar.sha1	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha1
maven-metadata.xml.sha256					performance-analyzer-commons-1.0-20230523.200158-1.jar.sha256	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha256
maven-metadata.xml.sha512					performance-analyzer-commons-1.0-20230523.200158-1.jar.sha512	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha512
```